src/main/java/org/jabref/gui/filelist/LinkedFilesEditDialogViewModel.java

```diff
@@ -0,0 +1,126 @@
+package org.jabref.gui.filelist;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import javafx.beans.property.ListProperty;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleListProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+import javafx.collections.FXCollections;
+
+import org.jabref.gui.AbstractViewModel;
+import org.jabref.gui.DialogService;
+import org.jabref.gui.externalfiletype.ExternalFileType;
+import org.jabref.gui.externalfiletype.ExternalFileTypes;
+import org.jabref.gui.externalfiletype.UnknownExternalFileType;
+import org.jabref.gui.util.FileDialogConfiguration;
+import org.jabref.logic.util.io.FileUtil;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.LinkedFile;
+import org.jabref.model.util.FileHelper;
+import org.jabref.preferences.JabRefPreferences;
+import org.jabref.preferences.PreferencesService;
+
+public class LinkedFilesEditDialogViewModel extends AbstractViewModel {
+
+    private static final Pattern REMOTE_LINK_PATTERN = Pattern.compile("[a-z]+://.*");
+    private final StringProperty linkProperty = new SimpleStringProperty("");
+    private final StringProperty descriptionProperty = new SimpleStringProperty("");
+    private final ListProperty<ExternalFileType> externalfilesTypes = new SimpleListProperty<>(FXCollections.emptyObservableList());
+    private final ObjectProperty<ExternalFileType> selectedExternalFileType = new SimpleObjectProperty<>();
+    private final BibDatabaseContext database;
+    private final DialogService dialogService;
+    private final PreferencesService preferences;
+
+    public LinkedFilesEditDialogViewModel(LinkedFile linkedFile, BibDatabaseContext database, DialogService dialogService, PreferencesService preferences2) {
+        this.database = database;
+        this.dialogService = dialogService;
+        this.preferences = preferences2;
+        externalfilesTypes.set(FXCollections.observableArrayList(ExternalFileTypes.getInstance().getExternalFileTypeSelection()));
+        setValues(linkedFile);
+    }
+
+    private void checkExtension() {
```

Comment:
rename to `setExtension`? I would also add the link as a parameter to make it clear that the method guesses the extension based on the path.